earthpy/tests/test_plot_draw_legend.py

```diff
@@ -30,169 +30,238 @@ def binned_array_3bins():
     plt.close(f)
 
 
-def test_listed_cmap(binned_array):
-    """Test that the the legend generates properly when provided with a Listed
-     colormap"""
+def test_listed_cmap(arr_plot_list_cmap):
+    """Test that the the legend generates properly when provided with a ListedColormap"""
 
-    bins, arr_class = binned_array
+    im_ax, arr = arr_plot_list_cmap
 
-    # TODO make the list of colors a fixture for reuse
-    cmap_list = ListedColormap(
-        ["white", "tan", "purple", "springgreen", "darkgreen"]
-    )
-    f, ax = plt.subplots()
-    im_plt = ax.imshow(arr_class, cmap=cmap_list)
-    leg = ep.draw_legend(im_plt)
+    leg = ep.draw_legend(im_ax)
     legend_cols = [i.get_facecolor() for i in leg.get_patches()]
-    assert len(legend_cols) == len(bins) - 1
-    plt.close(f)
+    assert len(legend_cols) == len(np.unique(arr))
+    plt.close()
 
 
-def test_noncont_listed_cmap(binned_array, listed_cmap):
-    """Test that an arr with 3 classes (missing the 1, and 5) which
-     would need to be normalized, only creates a legend with x handles
-     by default"""
+def test_classes_provided_as_array(arr_plot_list_cmap):
+    """Test that draw_legend works when classes are provided as an arr (not a list)."""
 
-    cmap, norm = listed_cmap
-    bins, arr_class = binned_array
+    im_ax, arr = arr_plot_list_cmap
+    n_classes = 5
 
-    arr_class[arr_class == 1] = 2
-    arr_class[arr_class == 5] = 4
+    leg = ep.draw_legend(im_ax, classes=np.arange(n_classes))
 
-    f, ax = plt.subplots(figsize=(5, 5))
-    im = ax.imshow(arr_class, cmap=cmap, norm=norm)
-    leg = ep.draw_legend(im)
+    legend_cols = [i.get_facecolor() for i in leg.get_patches()]
+    assert len(legend_cols) == n_classes
+    plt.close()
+
+
+def test_noncont_listed_cmap(vals_missing_plot_list_cmap):
+    """An arr with 3 vals (missing the 1, and 5) which requires normalization
+     produces creates a legend with 3 handles."""
+
+    im_ax, arr = vals_missing_plot_list_cmap
+
+    leg = ep.draw_legend(im_ax)
 
     legend_cols = [i.get_facecolor() for i in leg.get_patches()]
-    assert len(legend_cols) == len(np.unique(arr_class))
-    plt.close(f)
+    assert len(legend_cols) == len(np.unique(arr))
+    plt.close()
 
 
-def test_noncont_listed_cmap_3_classes(binned_array, listed_cmap):
-    """Test legend for a non continuous listed cmap where
+def test_listed_cmap_3_classes(vals_missing_plot_list_cmap):
+    """Test legend for a listed cmap where
     the user wants all classes to be drawn in the legend. IE the classified
     image has classes 2,3,4 and the user wants classes 1-5 to appear """
 
-    cmap, norm = listed_cmap
-    bins, arr_class = binned_array
+    im_ax, arr = vals_missing_plot_list_cmap
 
-    f, ax = plt.subplots(figsize=(5, 5))
-    im = ax.imshow(arr_class, cmap=cmap, norm=norm)
-    leg = ep.draw_legend(im, classes=[1, 2, 3, 4, 5])
+    leg = ep.draw_legend(im_ax, classes=[1, 2, 3, 4, 5])
```

Comment:
Instead of writing out `[1, 2, 3, 4, 5]` twice in this test, you could define that as an object, e.g., `class_list` below: 

```python
def test_listed_cmap_3_classes(vals_missing_plot_list_cmap):
    """Test legend for a listed cmap where
    the user wants all classes to be drawn in the legend. IE the classified
    image has classes 2,3,4 and the user wants classes 1-5 to appear """

    im_ax, arr = vals_missing_plot_list_cmap

    class_list = list(range(5))
    leg = ep.draw_legend(im_ax, classes=class_list)
    legend_cols = [i.get_facecolor() for i in leg.get_patches()]

    assert len(legend_cols) == len(class_list)
    plt.close()

```